src/patterns/ec2-app.ts

```diff
@@ -192,19 +194,21 @@ export class GuEc2App {
 
     const listener = new GuHttpsApplicationListener(scope, "Listener", {
       app,
+      loadBalancer,
+      certificate,
+      targetGroup,
       // When open=true, AWS will create a security group which allows all inbound traffic over HTTPS
       open: props.access.scope === AccessScope.PUBLIC,
-      loadBalancer: loadBalancer,
-      certificate: certificate,
-      targetGroup: targetGroup,
     });
 
     // Since AWS won't create a security group automatically when open=false, we need to add our own
     if (props.access.scope === AccessScope.RESTRICTED) {
-      listener.connections.addSecurityGroup(
-        new GuSecurityGroup(scope, "SecurityGroup", {
+      loadBalancer.connections.addSecurityGroup(
```

Comment:
Minor: I think we can drop the `.connections` here, as `ApplicationLoadBalancer` has an `addSecurityGroup` [method](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-ec2.Connections.html#addwbrsecuritywbrgroupsecuritygroups).

Reply:
Good catch - implemented this and tests & snapshots still pass so all good.